src/EnergyPlus/Autosizing/HeatingAirflowUASizing.cc

```diff
@@ -0,0 +1,217 @@
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <EnergyPlus/Autosizing/HeatingAirflowUASizing.hh>
+#include <EnergyPlus/Data/EnergyPlusData.hh>
+#include <EnergyPlus/DataEnvironment.hh>
+#include <EnergyPlus/api/TypeDefs.h>
+
+namespace EnergyPlus {
+
+    void HeatingAirflowUASizer::initializeWithinEP(EnergyPlusData &state,
+                                                   std::string const &_compType,
+                                                   std::string const &_compName,
+                                                   bool const &_printWarningFlag,
+                                                   std::string const &_callingRoutine) {
+
+        BaseSizer::initializeWithinEP(state, _compType, _compName, _printWarningFlag, _callingRoutine);
+        this->sizingString = "Heating Coil Airflow For UA"; // TODO: We should make this a constexpr so we don't need this overridden function in each class just for this one line
+    }
+
+
+    void HeatingAirflowUASizer::initializeForSingleDuctZoneTerminal(Real64 const elevation, Real64 mainFlowRate) {
+        this->initializeFromAPI(elevation);
+        this->zoneSizingRunDone = true;
+        this->curZoneEqNum = 1;
+        this->termUnitSingDuct = true;
+        this->curTermUnitSizingNum = 1;
+        this->termUnitSizing.allocate(1);
+        this->termUnitSizing(1).AirVolFlow = mainFlowRate;
+    }
+
+    void HeatingAirflowUASizer::initializeForZoneInductionUnit(Real64 const elevation, Real64 mainFlowRate,
+                                                               Real64 reheatMultiplier) {
+        this->initializeFromAPI(elevation);
+        this->zoneSizingRunDone = true;
+        this->curZoneEqNum = 1;
+        this->termUnitIU = true;
+        this->termUnitPIU = true; // probably don't need to set both, but whatever
+        this->curTermUnitSizingNum = 1;
+        this->termUnitSizing.allocate(1);
+        this->termUnitSizing(1).AirVolFlow = mainFlowRate;
+        this->termUnitSizing(1).ReheatAirFlowMult = reheatMultiplier;
+    }
+
+    void HeatingAirflowUASizer::initializeForZoneFanCoil(Real64 const elevation, Real64 designHeatVolumeFlowRate) {
+        this->initializeFromAPI(elevation);
+        this->zoneSizingRunDone = true;
+        this->zoneEqFanCoil = true;
+        this->curZoneEqNum = 1;
+        this->finalZoneSizing.allocate(1);
+        this->finalZoneSizing(1).DesHeatVolFlow = designHeatVolumeFlowRate;
+    }
+
+    // TODO: What would the zone otherTypeEq include?
+
+    void HeatingAirflowUASizer::initializeForSystemOutdoorAir(Real64 const elevation, Real64 overallSystemMassFlowRate,
```

Comment:
We need to decide what to ask from the user for this part.  I put in a "DOAS" flag for now, and then handle the initialization different based on that flag.  But in one side the code looks for a mass flow rate, and in the other side the code looks for a volume flow rate.  We should discuss.